### PR TITLE
Add a population statistic.

### PR DESCRIPTION
**Pull request by LGM-Doyle**

### CODE REVIEW

**Comment by Vezzra** [comment]
@LGM-Doyle, good idea, but why set this to mandatory for 0.4.8? Not that I expect any issues with it, but still, if there were, something like that shouldn't be release blocking...?

**Comment by LGM-Doyle** [comment]
Changed.

I did not mean it to be release blocking.  

I had spliced it off from the leak fixing PR #1838, which I think should be release blocking.  The leaks due to valid FOCS can cause memory exhaustion. For example, adding redundant parentheses.

File: default/scripting/empire_statistics/Empire.focs.txt [hunk 1]
```diff
@@ -24,3 +24,11 @@ Statistic name = "PLANETS_DEPOPULATED" value =
 
 Statistic name = "PLANETS_INVADED" value =
     SpeciesPlanetsInvaded empire = Source.Owner
+
+Statistic name = "TOTAL_POPULATION_STAT"
+    value = Statistic Sum
+        value = Source.Population
```
Comment (geoffthemedio) [inline default/scripting/empire_statistics/Empire.focs.txt]: I doubt you really mean / want `Source.Population`. Probably `LocalCandidate.Population`.

See: https://github.com/freeorion/freeorion/blob/master/universe/ValueRef.h#L903

**Comment by LGM-Doyle** [comment]
@geoffthemedio thanks for reviewing.

**Pull request merged**